trade.lic

```diff
@@ -1,78 +1,425 @@
-    @hometown = @settings.hometown
-    bput("open my #{@contract_container}", 'You open', 'That is already')
-    @equipment_manager.empty_hands
+    @hub_city = 'Crossing'
+    @emergency_delivery_town = nil
+    @current_caravan_destination = nil
+    @caravan_coins_on_hand = @settings.caravan_coins_on_hand.to_i
+    @caravan_interior = @settings.caravan_interior
+    @caravan_being_led = false
+    @force_contract_from = []
+    Flags.add('lead-arrived', 'having arrived at its destination', 'You feel the momentum of the room shift and then hear a voice from outside announce')
+    Flags.add('lead-failed', 'Your caravan guide comes to you and suggests')
+    Flags.add('caravan-arrived', /^Your .*, following you/)
+
+    #TRAINING
+    @gem_pouch_adjective = @settings.gem_pouch_adjective
+    @full_pouch_container = @settings.full_pouch_container
+    @training_spells = @settings.crafting_training_spells
+    @step_toggle = -1
+    wipe_token
+    Flags.add('trading-magic-ready', 'You feel fully prepared to cast your spell.')
+
+    @equipment_manager = EquipmentManager.new(@settings)
+    @cooldown_timers = {}
+    @caravan_training_skills = @settings.caravan_training_skills
+    @lead_training_skills = @caravan_training_skills.select { |skill, _cooldown| !['Mechanical Lore', 'Augmentation', 'Utility', 'Warding'].include?(skill) }
+    @walk_training_skills = @caravan_training_skills.select { |skill, _cooldown| ['Augmentation', 'Utility', 'Warding', 'Attunement', 'Performance', 'Appraisal', 'Mechanical Lore'].include?(skill) }
+    @craft_bag = @settings.crafting_container
+    @craft_belt = @settings.engineering_belt
+
+    @yarn_quantity = @settings.yarn_quantity > 300 ? 301 : @settings.yarn_quantity
+    @lumber_quantity = @settings.lumber_quantity > 30 ? 30 : @settings.lumber_quantity
+    @current_yarn_count = 0
+    @current_lumber_count = 0
+    @current_grass_count = 100
+    @times_outfitting = 0
+    @times_engineering = 0
+
+    @box = @settings.picking_lockbox
+    @worn_lockbox = @settings.picking_worn_lockbox
+    @pet_box_source = @settings.picking_pet_box_source
+    @loot_nouns = @settings.lootables
+    @trash_nouns = get_data('items').trash_nouns
+
+    arg_definitions = [
+      [
+        { name: 'destination', regex: /\d+/i, optional: true, description: 'Give a room ID to bring your caravan to' }
+      ],
+      [
+        { name: 'closeup', regex: /closeup/i, optional: false, description: 'Close up by delivering all remaining crates and accepting no new contracts' },
+        { name: 'exit', regex: /exit/i, optional: true, description: 'Exits the game after delivering all crates and returning the caravan' }
+      ],
+      [
+        { name: 'outpost', regex: /outpost/i, optional: false, description: 'Bring the caravan to the trade outpost of a town.' },
+        { name: 'town_name', regex: /\w+/i, variable: true, description: 'First word of the town name.' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
```

Comment:
This is nitpicking but arg parsing could be done first; if the users args dont make sense then there's no reason to do any of the above initialization steps..